pkg/webhook/bootstrap.go

```diff
@@ -76,89 +81,125 @@ func (s *Server) setBootstrappingDefault() {
-	}
+	objects := []runtime.Object{s.mutatingWebhookConfig, s.validatingWebhookConfig, svc}
 
-	err = createOrReplace(s.Client, s.mutatingWebhookConfig, func(current, desired runtime.Object) error {
-		typedC := current.(*admissionregistration.MutatingWebhookConfiguration)
-		typedD := desired.(*admissionregistration.MutatingWebhookConfiguration)
-		typedC.Webhooks = typedD.Webhooks
-		return nil
-	})
-	if err != nil {
-		return nil, err
+	for _, obj := range objects {
+		err = createOrReplace(s.Client, obj)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
+}
 
-	err = createOrReplace(s.Client, s.validatingWebhookConfig, func(current, desired runtime.Object) error {
-		typedC := current.(*admissionregistration.ValidatingWebhookConfiguration)
-		typedD := desired.(*admissionregistration.ValidatingWebhookConfiguration)
-		typedC.Webhooks = typedD.Webhooks
-		return nil
-	})
-	if err != nil {
-		return nil, err
-	}
+type mutateFn func(current, desired runtime.Object) error
 
-	return nil, nil
+var serviceFn = func(current, desired runtime.Object) error {
+	typedC := current.(*corev1.Service)
+	typedD := desired.(*corev1.Service)
+	typedC.Spec.Selector = typedD.Spec.Selector
+	return nil
 }
 
-type mutateFn func(current, desired runtime.Object) error
+var mutatingWebhookConfigFn = func(current, desired runtime.Object) error {
+	typedC := current.(*admissionregistration.MutatingWebhookConfiguration)
+	typedD := desired.(*admissionregistration.MutatingWebhookConfiguration)
+	typedC.Webhooks = typedD.Webhooks
+	return nil
+}
+
+var validatingWebhookConfigFn = func(current, desired runtime.Object) error {
+	typedC := current.(*admissionregistration.ValidatingWebhookConfiguration)
+	typedD := desired.(*admissionregistration.ValidatingWebhookConfiguration)
+	typedC.Webhooks = typedD.Webhooks
+	return nil
+}
 
 // createOrReplace creates the object if it doesn't exist;
 // otherwise, it will replace it.
-// When replacing, fn knows how to preserve existing fields in the object GET from the APIServer.
+// When replacing, it knows how to preserve existing fields in the object GET from the APIServer.
+func createOrReplace(c client.Client, obj runtime.Object) error {
+	switch obj.(type) {
+	case *admissionregistration.MutatingWebhookConfiguration:
+		return createOrReplaceHelper(c, obj, mutatingWebhookConfigFn)
+	case *admissionregistration.ValidatingWebhookConfiguration:
+		return createOrReplaceHelper(c, obj, validatingWebhookConfigFn)
+	case *corev1.Service:
+		return createOrReplaceHelper(c, obj, serviceFn)
+	}
+	return nil
+}
+
+// createOrReplaceHelper creates the object if it doesn't exist;
+// otherwise, it will replace it.
+// When replacing, fn  should know how to preserve existing fields in the object GET from the APIServer.
 // TODO: use the helper in #98 when it merges.
-func createOrReplace(c client.Client, obj runtime.Object, fn mutateFn) error {
+func createOrReplaceHelper(c client.Client, obj runtime.Object, fn mutateFn) error {
```

Comment:
hmm....looks like this could be useful in general. We might be able to move these under `controller-runtime` under some `clientutil` pkg.

Reply:
This may be a temporary thing.
#98 will land sometime.